Specs/TerrainTileProcessor.js

```diff
@@ -0,0 +1,123 @@
+define([
+    'Core/clone',
+    'Renderer/Texture',
+    'Scene/GlobeSurfaceTile',
+    'Scene/ImageryLayer',
+    'Scene/TerrainState',
+    'ThirdParty/when'
+], function(
+    clone,
+    Texture,
+    GlobeSurfaceTile,
+    ImageryLayer,
+    TerrainState,
+    when) {
+    'use strict';
+
+    function TerrainTileProcessor(frameState, terrainProvider, imageryLayerCollection) {
+        this.frameState = frameState;
```

Comment:
This is a mock class for testing right? How come it's not called `MockX` in the same way the others like `MockTerrainProvider` and `MockImageryProvider` are named?